structuredheadings/plugin.js

```diff
@@ -1,35 +1,72 @@
 (function () {
-	CKEDITOR.plugins.add('structuredheadings', {
-		icons: 'structuredheadings',
-		init: (editor) => {
-			// The command to add an autonumbering class
-			editor.addCommand('autoNumberHeading', {
-				exec: (editor) => {
-					/* Get all header elements and assign classes
-					 * only if needed
-					 */
-					assignClassToList(editor.document.find('h1,h2,h3'));
-				}
-			});
-			
-			// Adding the button to the toolbar
-			editor.ui.addButton( 'structuredheadings', {
-				label: 'Autonumber Heading',
-				command: 'autoNumberHeading',
-				toolbar: 'styles,0'
-			})
-			return editor; // actually, no return is required, but this shuts eslint up
-		}
-	});
-	
-	function assignClassToList(list) {
-		// Iterate over the list of nodes
-		for (var i = 0; i < list.count(); i++) {
-			var h = list.getItem(i);
-			// Add the class if it doesn't have it
-			if(!h.hasClass('autonumber')) {
-				h.addClass('autonumber');
-			}
-		};
-	}
+  CKEDITOR.plugins.add("structuredheadings", {
+    icons: "structuredheadings",
+    init: function (editor) {
+      // list of elements allowed to be numbered
+      var allowedElements = ["h1", "h2", "h3", "h4", "h5", "h6"];
+
+      editor.addContentsCss(this.path + "styles/numbering.css");
+
+      // Helper function to set button state
+      var setState = function (state) {
+        switch (state) {
+        case "on":
```

Comment:
I'm guessing there might be an eslint warning here (I haven't looked at the output yet)